src/mlpack/tests/neat_test_tools.hpp

```diff
@@ -197,79 +213,80 @@ class DPNVTask
-
-        // Scale the input between -1 and 1.
-        input[0] /= 2.4;
-        input[1] /= 36 * 2 * 3.1416 / 360;
-        input[2] /= 36 * 2 * 3.1416 / 360;
-        
-        wiggleBuffer1.push_back(std::abs(inputMatrix(0,0)) + std::abs(inputMatrix(0,1)) +
-            std::abs(inputMatrix(1,0)) + std::abs(inputMatrix(1,1)));
-        
-        if (wiggleBuffer1.size() == 100)
-        {
-          wiggleBuffer2.push_back(wiggleBuffer1.front());
-          wiggleBuffer1.pop_front();
-        }
-        if (wiggleBuffer2.size() == 100)
-          wiggleBuffer2.pop_front();
-
-        output = genome.Evaluate(input);
-
-        if (timeStep > 499)
-        {
-          double sum = 0;
-          for (auto it = wiggleBuffer2.begin(); it != wiggleBuffer2.end(); it++)
-            sum += *it;
-          if (sum > 30)
-            break;
-        }
-      }
+      // Use the current action to get the next state.
+      environment.Sample(state, action, state);
 
-      if(timeStep > 499 && timeStep < 600)
+      // Update the state of the genome for the next step.
+      inputMatrix = state.Data();
+      input = {inputMatrix(0, 0), inputMatrix(0, 1), inputMatrix(0, 2)};
+
+      // Scale the input between -1 and 1.
+      input[0] /= 2.4;
+      input[1] /= 36 * 2 * 3.1416 / 360;
+      input[2] /= 36 * 2 * 3.1416 / 360;
+      
+      wiggleBuffer1.push_back(std::abs(inputMatrix(0,0)) + std::abs(inputMatrix(0,1)) +
+          std::abs(inputMatrix(1,0)) + std::abs(inputMatrix(1,1)));
+      
+      if (wiggleBuffer1.size() == 100)
       {
-        double sum = 0;
-        for (auto it = wiggleBuffer1.begin(); it != wiggleBuffer1.end(); it++)
-          sum += *it;
-        fitness += timeStep + (10.0 / std::max(1.0, sum));
+        wiggleBuffer2.push_back(wiggleBuffer1.front());
+        wiggleBuffer1.pop_front();
       }
-      else if(timeStep > 599)
+      if (wiggleBuffer2.size() == 100)
+        wiggleBuffer2.pop_front();
+
+      output = genome.Evaluate(input);
+
+      if (timeStep > 499)
       {
         double sum = 0;
         for (auto it = wiggleBuffer2.begin(); it != wiggleBuffer2.end(); it++)
           sum += *it;
-        fitness += timeStep + (10.0 / std::max(1.0, sum));
+        if (sum > 30)
+          break;
       }
-      else
-        fitness += timeStep;
     }
 
-    std::cout <<" Fitness: " << fitness/20 << std::endl;
-    return fitness / 20;
+    if(timeStep > 499 && timeStep < 600)
+    {
+      double sum = 0;
+      for (auto it = wiggleBuffer1.begin(); it != wiggleBuffer1.end(); it++)
+        sum += *it;
+      fitness += timeStep + (10.0 / std::max(1.0, sum));
+    }
+    else if(timeStep > 599)
+    {
+      double sum = 0;
+      for (auto it = wiggleBuffer2.begin(); it != wiggleBuffer2.end(); it++)
+        sum += *it;
+      fitness += timeStep + (10.0 / std::max(1.0, sum));
+    }
+    else
+      fitness += timeStep;
+
+    std::cout <<" Fitness: " << fitness << std::endl;
```

Comment:
In case we are going to provide such a function in the final version, we should go for ``Log::Debug``instead of ``std::cout``.